### ISSUE TITLE
TimeoutError in headless mode

### ISSUE THREAD

**Issue by jmbowman** [pa11y/pa11y#432]
I'm hitting timeout errors with pa11y on pages requiring authentication, even with the correct headers specified in the config file, but *only* when Chrome launches in the default headless mode.  If I add `"headless": false` to `chromeLaunchConfig` in the configuration file, it works correctly (but runs slower, requires xvfb for CI on Linux, etc.)  I'm trying to upgrade a test suite that currently uses pa11y 4.0.1 and phantomjs to use pa11y 5.0.4 and headless Chrome instead.  Pages which don't require authentication are loading as expected.  It doesn't seem to matter whether I use the Chromium downloaded by pa11y or the system's Chrome installation (I've tried both).

## Expected behaviour

```bash
./node_modules/.bin/pa11y http://localhost:18003/courses/course-v1:edX+Test101+course/courseware/e3f904e672574a32a16328061041bf65/7bce26c3db5549f7adb7094ea9529b10/ --config=pa11y.json --include-notices --include-warnings --reporter=json
[{"code":"WCAG2AA.Principle2.Guideline2_4.2_4_2.H25.2","context":"<title>\n...
```

## Actual behaviour

```bash
./node_modules/.bin/pa11y http://localhost:18003/courses/course-v1:edX+Test101+course/courseware/e3f904e672574a32a16328061041bf65/7bce26c3db5549f7adb7094ea9529b10/ --config=pa11y.json --include-notices --include-warnings --reporter=json
TimeoutError: Pa11y timed out (30000ms)
    at Timeout.setTimeout [as _onTimeout] (/Users/jeremybowman/Source/edX/pa11ycrawler/node_modules/p-timeout/index.js:27:54)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)
```

## Steps to reproduce

I haven't tried creating a minimal test case yet, but the failure in CI with logs, etc. can be found from https://github.com/edx/edx-platform/pull/19120 .  The rest of the code changed to handle the pa11y upgrade in this project can be found in a utility package at https://github.com/edx/pa11ycrawler/pull/84

## Environment:
```
Pa11y:      5.0.4
Node.js:    8.9.3
npm:        5.5.1
OS:         17.7.0 (darwin)
```

Also:

```
Pa11y:      5.0.4
Node.js:    8.9.3
npm:        5.5.1
OS:         4.9.93-linuxkit-aufs (linux)
```

Configuration file example:

```javascript
{
  "headers": {
    "Accept-Language": "en",
    "Accept-Encoding": "gzip,deflate",
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
    "User-Agent": "Scrapy/1.1.2 (+http://scrapy.org)",
    "Referer": "http://localhost:18003/courses/course-v1:edX+Test101+course/courseware/e3f904e672574a32a16328061041bf65/7bce26c3db5549f7adb7094ea9529b10/",
    "Cookie": "csrftoken=OIt5KxnhjZE2HeMS0oq27Y51liBt7m6ySRDDY7EuwnUvMgrEEBfqgRddsDWwBhce; sessionid=\"1|orvxn14oml958udrz5vusr27upnrpsyr|hEwcC2JfxNZq|IjEwYTQ5OTExZGI0NWFhMTlhNTliNTRkMzIwYjRkYzE2N2FkNTY2Y2FiM2VmYjEyMzRiNDUzYjVhNmMxYWY0MWEi:1gCWMW:iQ17RMWFfyeGkBC5DH64MOqCg3U\""
  },
  "chromeLaunchConfig": {
    "args": ["--no-sandbox", "--disable-setuid-sandbox"]
  }
}
```

**Comment by jmbowman** [comment]
I noticed that pa11y 5.1.0 was just released, and tried that also; the same problem persists, unfortunately.

**Comment by josebolos** [comment]
Hi @jmbowman!

Thanks a lot for the detailed report, and apologies for not being able to get back to you sooner.

While investigating #570 I've found at least 4 different instances of Chrome either failing to open when launched from puppeteer, or randomly disconnecting without apparent reason while open. I'm wondering if one or more of these issues may be what was causing the problem that you reported.

The good news is that we've just released v6.0.0 of pa11y which includes the latest version of puppeteer. This version fixes a more recent version of Chrome and fixes many of the issues that the `1.x` branch of puppeteer that we use in pa11y 5 has. If you're still experiencing this issue it may be worth trying with pa11y 6 and seeing if it helps in any way.

Thanks again for the bug report!
